pyuvdata/tests/__init__.py

```diff
@@ -5,19 +5,245 @@
+                )
+
+                for warn_i, exp_warn in enumerate(self.expected_warning):
+                    if not any(issubclass(r.category, exp_warn) for r in self):
+                        __tracebackhide__ = True
+                        raise AssertionError(
+                            "DID NOT WARN. No warnings of type {} was emitted. "
+                            "The list of emitted warnings is: {}.".format(
+                                self.expected_warning, [each.message for each in self]
+                            )
+                        )
+                    elif self.match is not None:
+                        for record in self:
+                            if str(record.message).startswith(
+                                self.match[warn_i]
+                            ) or re.compile(self.match[warn_i]).search(
+                                str(record.message)
+                            ):
+                                if issubclass(
+                                    record.category, self.expected_warning[warn_i]
+                                ):
+                                    break
+                        else:
+                            raise AssertionError(
+                                f"No warnings of type {self.expected_warning[warn_i]} "
+                                f"matching ('{self.match[warn_i]}') was "
+                                "emitted. The list of emitted warnings is: "
+                                f"{[each.message for each in self]}."
+                            )
+
+
+def check_warnings(expected_warning, match, nwarnings=None, *args, **kwargs):
+    """
+    Assert that code raises a particular set of warnings, used as a context manager.
+
+    Similar to ``pytest.warns``, but allows for specifying multiple warnings.
+    It also better matches warning strings when the warning uses f-strings or
+    formating.
+
+    Note that unlike the older checkWarnings function, the warnings can be passed
+    in any order, they do not have to match the order the warnings are raised
+    in the code.
+
+    Parameters
+    ----------
+    expected_warning : list of Warnings or Warning
+        List of expected warnings. If a single warning type or a length 1 list,
+        will be used for the type of all warnings.
+    match : list of str or str
+        List of strings to match warnings to. If a str or a length 1 list, will
+        be used for all warnings.
+    nwarnings : int, optional
+        Option to specify that multiple of a single type of warning is expected.
+        Only used if category and match both only have one element.
+
+    """
+    __tracebackhide__ = True
+    if not isinstance(expected_warning, list):
+        expected_warning_list = [expected_warning]
+    else:
+        expected_warning_list = expected_warning
+    if not isinstance(match, list):
+        match_list = [match]
+    else:
+        match_list = match
```

Comment:
Should we use the `list()` function here to cast as a list? What if the inputs are tuples, sets, or numpy arrrays?

Reply:
or perhaps `list(uvutils._get_iterable())` to ensure a single string is not broken up?

Reply:
Oh that wouldn't necessarily work either, might need a special case of `isinstance( ???, str)` if you do what I suggested.

Reply:
hmm, I should probably be more careful. I was expecting to get either a string or a list. I can add some checking for that.